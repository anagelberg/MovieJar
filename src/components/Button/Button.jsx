import './Button.scss';

function Button({ modifier, text, icon, ...buttonAttributes }) {
    return (
        <button className={`cta cta${modifier}`} {...buttonAttributes}>
            <div className="cta__contents">
                {icon && <img className="cta__icon" src={icon} alt="button icon"></img>}
                {text}
            </div>
        </button>
    );
}

export default Button;