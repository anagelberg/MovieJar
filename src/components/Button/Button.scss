@use "../../styles/variables" as *;
@use "../../styles/mixins" as *;

.cta {
  @include flex-center;
  align-self: flex-end;
  height: $field-input-height;
  width: 100%;
  border-radius: $border-radius;
  color: $dark-text;
  background-color: $accent;
  transition: $transition;
  border: none;

  &__contents {
    @include flex-center;
    position: relative;
    align-items: center;
    width: 100%;
    height: 100%;
  }

  &__icon {
    position: absolute;
    left: $icon-spacing-unit;
  }

  @include tablet {
    max-width: 8rem;
  }

  &:disabled {
    background-color: $secondary-background;
  }

  &:hover:enabled {
    background-color: $primary;
    cursor: pointer;
  }

  &:hover:disabled {
    cursor: not-allowed;
  }

  &--reverse {
    border: 1px solid $secondary-background;
    background-color: $light-background;
    color: $dark-text;

    &:hover:enabled {
      background-color: $secondary-background;
      color: $light-text;
    }
  }

  &--delete {
    background-color: $error;
    color: $light-text;
    border: 1px solid $error;

    &:hover:enabled {
      background-color: $light-background;
      color: $error;
    }
  }
}
