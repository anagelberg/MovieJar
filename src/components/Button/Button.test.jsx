import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from './Button';
import TestIcon from '../../assets/icons/google.svg';

test('renders Button component', () => {
    render(<Button />);
    expect(screen.getByRole('button')).toBeInTheDocument();
});

test('renders Button component with text', () => {
    render(<Button text="test text" />);
    expect(screen.getByText('test text')).toBeInTheDocument();
});

test('toggles Button modifier class', () => {
    const { container } = render(<Button modifier="--testing-class" />);
    expect(container.firstChild).toHaveClass('cta--testing-class')
});

test('Button calls clickHandler on click', () => {
    const clickHandler = jest.fn();
    render(<Button onClick={clickHandler} />);
    userEvent.click(screen.getByRole('button'));
    expect(clickHandler).toHaveBeenCalled();
});

test('Button is passed undefined props as default attributes', () => {
    render(<Button disabled={true} />);
    expect(screen.getByRole('button')).toBeDisabled();
});


test('Button component matches snapshot', () => {
    const { container } = render(<Button icon={TestIcon} />);
    expect(container.firstChild).toMatchSnapshot();
});