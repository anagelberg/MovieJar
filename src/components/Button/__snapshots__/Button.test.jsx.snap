// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<button
  class="cta ctaundefined"
>
  <div
    class="cta__contents"
  >
    <img
      alt="button icon"
      class="cta__icon"
      src="google.svg"
    />
  </div>
</button>
`;
